### DISCUSSION TITLE
Backend scalability

### DISCUSSION THREAD

**Discussion by zanzlender** [vercel/next.js#30831] [Help]
Hi everyone, and first of all sorry if this isn't the right place to ask... However I am really curious and cannot find answers elsewhere.

I'm a junior developer and I have been using Next for the past 5 months or so. Up until now I was working on smaller projects so it didn't really matter that much... but now as I'm starting on a project that has the potential to grow a lot, I want to think about scalability before I even start.

But enough about that, my question is, how good is Next's backend regarding scalability and how does it work exactly? What are it's bottlenecks?

Let me elaborate. I know Next can be used for enterprise-sized commercial projects, like webshops. However, what I want to know is how much of a load the backend, or serverless function, or APIs, can handle (numbers pls 👏), and how we would go about scaling it?
Is it as simple as: lets buy more servers, make copies of the production version, put a load balancer up front and we can scale to infinity? Or do we have to do it another way? 
Lets disregard the server and database bottlenecks for now... 

Thanks in advance 😁

**Comment by romanellius** [comment]
Hi, @[zanzlender](https://github.com/zanzlender) 👋

Didn't find Vercel team to publish this information officially but they run backends on **AWS Lambda** (serverless functions).

There are some proofs:
https://twitter.com/vercel/status/836665717344907264?s=20&t=VzZqd0JhyprG-uxLVEqpSQ
https://vercel.com/support/articles/how-can-i-improve-serverless-function-lambda-cold-start-performance-on-vercel

It means auto scalability but there're limits provided by AWS and Vercel (your current plan):
https://vercel.com/docs/concepts/limits/overview

PS/ sorry for the necropost

**Comment by two10** [comment]
I also have a similar question , i have 10 API written in the as next js pages/api , how much API per second i can hit on that
